### PR TITLE
[stable/postgersql] Allow enable metrics sidecar for slaves pods

### PR DESCRIPTION
**Pull request by kel00s**
#### What this PR does / why we need it:
Scrapping metrics by Prometheus can be possible for Slave pods.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:
@cedricgc 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] title of the PR contains starts with chart name e.g. `[stable/chart]`

### CODE REVIEW

File: stable/postgresql/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: postgresql
-version: 3.18.4
+version: 3.18.6
```
Comment (carrodher) [inline stable/postgresql/Chart.yaml:3]:
```suggestion
version: 3.18.5
```

File: stable/postgresql/values.yaml [hunk 2]
```diff
@@ -299,6 +299,7 @@ readinessProbe:
 ##
 metrics:
   enabled: false
+  slaveEnalbed: false
```
Comment (carrodher) [inline stable/postgresql/values.yaml]:
```suggestion
  slaveEnabled: false
```

**Review by carrodher — CHANGES_REQUESTED** [review]
Thank for the PR, please see my comments

File: stable/postgresql/Chart.yaml (hunk 1 above)
Comment (kel00s) [reply stable/postgresql/Chart.yaml:3]: I've opened another PR where I'm bumping-up the version to 5.

File: stable/postgresql/values.yaml (hunk 2 above)
Comment (kel00s) [reply stable/postgresql/values.yaml]: Done.

**Review by juan131 — CHANGES_REQUESTED** [review]
The metrics containers you're creating won't be accessible through the metrics service, since currently the **metric-svc.yaml** is using the selector below:

```yaml
  selector:
    app: {{ template "postgresql.name" . }}
    release: {{ .Release.Name }}
    role: master
```

You need to update those labels for the service to redirect traffic to them:

```diff
- {{- if .Values.metrics.enabled  }}
+ {{- if or .Values.metrics.enabled .Values.metrics.slaveEnabled }}
apiVersion: v1
kind: Service
...
spec:
  ...
  selector:
    app: {{ template "postgresql.name" . }}
    release: {{ .Release.Name }}
+   {{- if .Values.metrics.enabled  (not .Values.metrics.slaveEnabled) }}
    role: master
+   {{- else if .Values.metrics.slaveEnabled  (not .Values.metrics.enabled) }}
    role: slave
+   {{- end }}
```

**Pull request closed without merging**